### ISSUE TITLE
update not working?

### ISSUE THREAD

**Issue by tony-kerz** [feathersjs/feathers-sequelize#125]
### Steps to reproduce

submit http `PUT` for a resource passing body to be used for update operation

### Expected behavior
identified resource should be updated

### Actual behavior
`PUT` returns updated version of object, but no call to database is made leaving persisted resource unchanged

```
  app:http:steps http-update: method='PUT', url='http://localhost:3000/alerts/3', data={ username: 'user-3', folder: '/folder-3', file: 'file-3', bytes: 4, message: 'message-4' }, headers=undefined +0ms
  feathers:rest REST handler calling `update` from `/alerts/3` +74ms
Executing (default): SELECT `id`, `username`, `folder`, `file`, `bytes`, `message`, `createdAt`, `updatedAt` FROM `Alerts` AS `Alert` WHERE `Alert`.`id` = '3';
```

> `PATCH` seems to be working correctly...

> i stepped thru in a debugger, but wasn't sure exactly what to look for, if helpful, i can do so again with some guidance as to what to check into...

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working): 
feathers: `2.1.3`
feathers-rest: `1.7.3`
feathers-sequelize: `2.0.0`

**NodeJS version**: `8.1.0`

**Operating System**: `osx`

> fwiw: going against mysql `5.7.18` executing in a [cucumber-js](https://github.com/cucumber/cucumber-js) test...

**Comment by zusamann** [comment]
Can you post your code here?

**Comment by tony-kerz** [comment]
so the client is a cucumber test, but it is just submitting an http request like:
```
PUT:/alerts/3
```
with a body of:
```
{
  username: 'user-3',
  folder: '/folder-3',
  file: 'file-3',
  bytes: 4,
  message: 'message-4'
}
```
the sequelize model looks like:
```
export default function(sequelize, DataTypes) {
  return sequelize.define('Alert', {
    username: {
      type: DataTypes.STRING,
      allowNull: false
    },
    folder: {
      type: DataTypes.STRING,
      allowNull: false
    },
    file: {
      type: DataTypes.STRING,
      allowNull: false
    },
    bytes: DataTypes.INTEGER,
    message: {
      type: DataTypes.TEXT,
      allowNull: false
    }
  })
}
```
what seem to be relevant initialization steps looks like this:
```
  await sequelize.sync()
  const app = feathers()
  app.configure(feathersRest())
  app.use(bodyParser.json())
  app.use(bodyParser.urlencoded({extended: true}))

  const paginate = {default: 10, max: 200}

  _.each(models, (val, key) => {
    const path = `/${_.kebabCase(key)}s`
    dbg('exposing model=%o as path=%o', key, path)
    app.use(path, feathersSqlzService({Model: models[key], paginate}))
  })
```

**Comment by sicruse** [comment]
@tony-kerz does the operation work if you have a before hook which sets sequelize: {raw: false} ?

e.g. 
```
    update: [ hook => { hook.params.sequelize = { raw: false }; return hook; } ],
```

**Comment by tony-kerz** [comment]
i'm not familiar with that hook syntax.
is this equivalent?
```
export default function(sequelize, DataTypes) {
  return sequelize.define(
    'Alert',
    {
      username: {
        type: DataTypes.STRING,
        allowNull: false
      },
      folder: {
        type: DataTypes.STRING,
        allowNull: false
      },
      file: {
        type: DataTypes.STRING,
        allowNull: false
      },
      bytes: DataTypes.INTEGER,
      message: {
        type: DataTypes.TEXT,
        allowNull: false
      }
    },
    {
      hooks: {
        beforeUpdate: (data, options) => {
          options.raw = false
        }
      }
    }
  )
}
```
> fwiw, the above yields the same behavior

**Comment by sicruse** [comment]
@tony-kerz no not quite. the hook I refer too is [a feathers hook](https://docs.feathersjs.com/api/hooks.html). Looking at your sample code you can introduce it something like the following:

```
  _.each(models, (val, key) => {
    const path = `/${_.kebabCase(key)}s`
    dbg('exposing model=%o as path=%o', key, path)
    app.use(path, feathersSqlzService({Model: models[key], paginate}))
    app.service(path).hooks({
      before: {
        update: [ hook => { hook.params.sequelize = { raw: false }; return hook; } ]
      }
    })
  })
```

**Comment by andrewmunro** [comment]
@sicruse I have this same issue after upgrading from feathers-sequelize 1.4.5 to 2.0.0.

I've tracked it down to the `raw` option in sequelize as you suggested:

![image](https://user-images.githubusercontent.com/1107388/27346706-e058db58-55e5-11e7-8dab-8c2cdebd60a0.png)

This option means that no properties are registered as having changed, so it doesn't bother to do the sql query.

I don't really understand what the underlying issue is. Is this an issue with feathers-sequelize or with sequelize, it seems incorrect for the end application having to add a hook that sets the raw query option to false.

**Comment by andrewmunro** [comment]
Seems related to #99

**Comment by joeblubaugh** [comment]
I've been able to work around this issue by downgrading to v1.4.5

**Comment by tony-kerz** [comment]
thanks @sicruse, i can confirm that the following works around the issue:
```
yarn add feathers-hooks
```
```
import feathersHooks from 'feathers-hooks'
//...
  app.configure(feathersHooks())
//...
  _.each(models, (val, key) => {
    const path = `/${_.kebabCase(key)}s`
    dbg('exposing model=%o as path=%o', key, path)
    app.use(path, feathersSqlzService({Model: models[key], paginate}))
    app.service(path).hooks({
      before: {
        update: [
          hook => {
            hook.params.sequelize = {raw: false}
            return hook
          }
        ]
      }
    })
  })
```
it would be great to be able to skip this magic incantation for general use tho 👍

**Comment by chasenlehara** [comment]
Instead of downgrading to 1.x, I was able to make update work by following the second fix in this comment: https://github.com/feathersjs/feathers-sequelize/pull/106#issuecomment-298775970

**Comment by joeblubaugh** [comment]
That bums me out - the only way to actually have the db work as intended in 2.0.0 is to add a custom hook to all your write calls?

**Comment by marshallswain** [comment]
@joeblubaugh no, you maybe missed the first option:

```
app.use('/people', service({
    Model,
    raw: false
}));
```

Declaring `raw:false` in your Model prevents the need to use a hook.

**Comment by snewell92** [comment]
Thanks @marshallswain - just had an UPDATE not commit my changes to the db and this is what I needed when declaring my services.

In the docs it mentions to [get the sequelize instance](https://docs.feathersjs.com/api/databases/sequelize.html#working-with-sequelize-model-instances) to set raw to false, but it says `raw: true` is recommended. Can you point me to more documentation regarding what the raw option is? In sequelize, my [understanding of raw queries](http://docs.sequelizejs.com/manual/tutorial/raw-queries.html) is basically just prepared statements...? Is it the same here in feathers-sequelize land?

**Comment by joeblubaugh** [comment]
@snewell92 the `raw` option in feathers world means that the data object returned by the database adapter is a "plain" JavaScript object. It's not related to the database query itself. Sequelize works in part by attaching observables to a number of object properties in order to decide whether it needs to execute a database query.

Other feathers hooks seem to have a hard time operating on Sequelize instances - the workaround suggested here is to always use `raw` instances in the service, which will break many hooks. A more complete workaround is in the docs here:
https://docs.feathersjs.com/api/databases/sequelize.html#working-with-sequelize-model-instances

This involves converting between the Sequelize representation of the data and a "plain" JavaScript Object representation depending on what is needed for the hooks you're using. So when using Feathers with Sequelize, you have to be very mindful of the data representation required by each hook.

**Comment by daffl** [comment]
Fixed via #162 in v2.3.1
